File: Logging.Server/Entitys/LogEntity.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;

namespace Logging.Server
{
    public class LogEntity
    {

        public ObjectId _id { get; set; }

        public string Title { get; set; }

        public string Message { get; set; }

        public long IP { get; set; }

        public LogLevel Level { get; set; }

        public int AppId { get; set; }

        public long Time { get; set; }

        public string Source { get; set; }

        public int Thread { get; set; }

        public List<string> Tags { get; set; }
    }


    public class LogEntityComparer : IComparer<LogEntity>
    {
        public int Compare(LogEntity x, LogEntity y)
        {
            if (x == null && y == null) return 0;
            if (x == null) return 1;
            if (y == null) return -1;

            if (x.Time > y.Time) return -1;
            if (x.Time < y.Time) return 1;

            if (x.Time == y.Time)
            {
                if (x._id.Increment > y._id.Increment) return -1;
                if (x._id.Increment < y._id.Increment) return 1;
            }

            return 0;
        }
    }
}

File: client/go/loggingclient/LogEntity.go
package loggingclient

type LogEntity struct {
	Type int

	Title string

	Message string

	Level int8

	Time int64

	Source string

	Thread int32

	Tags map[string]string
}

func NewLogEntity() *LogEntity {

	var logEntity = new(LogEntity)
	logEntity.Type = 1
	logEntity.Thread = 0 //Thread.CurrentThread.ManagedThreadId
	return logEntity
}

func (this *LogEntity) getType() int {
	return 1
}
